src/config/dnsmasq_config.c

```diff
@@ -345,7 +345,7 @@ bool __attribute__((const)) write_dnsmasq_config(struct config *conf, bool test_
 	{
 		fputs("# Enable query logging\n", pihole_conf);
 		if(conf->misc.extraLogging.v.b)
-			fputs("log-queries=extra\n", pihole_conf);
+			fputs("log-queries=proto\n", pihole_conf);
```

Comment:
What does this change?

Reply:
It adds the protocol used to perform a query. This is a new feature as `dnsmasq` is now able to perform cross-protocol queries when needed, e.g., UDP queries that were truncated due to the packets size. This is particularly an issue with DNSSEC being enabled and extra key information is attached. So far, `dnsmasq` refused to do such queries and sent `TRUNCATED` downstream to the client. Now, `dnsmasq` is able to do a query itself using TCP (no truncation can happen because of allowed packet fragmentation) and later sent the result down to the client again using UDP.

Example log output:
```
                                   |||
                                   vvv
Dec  21 16:31:31 dnsmasq[2281585]: UDP 436 127.0.0.1/36363 query[PTR] 1.2.168.192.in-addr.arpa from 127.0.0.1
```